sdk/dotnet/Pulumi/Deployment/DeploymentOutputs.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Collections.Generic;
+
+namespace Pulumi
+{
+    /// <summary>
+    /// A type-safe wrapper around <see cref="IDictionary{String, Object}"/> for generating stack outputs from created resources. 
+    /// </summary>
+    public class DeploymentOutputs
+    {
+        private readonly Dictionary<string, object?> outputs;
+
+        private DeploymentOutputs()
+        {
+            outputs = new Dictionary<string, object?>();
+        }
+
+        public static DeploymentOutputs Create()
+        {
+            return new DeploymentOutputs();
+        }
+
+        public void Export<T>(string outputName, Output<T> value)
```

Comment:
This still allows passing an arbitrary object as a value, right?

Reply:
@mikhailshilkov You were right: any arbitrary value could be implicitly converted to `Output<T>` but now I've removed this from the implementation. Only a select bunch of types can be used with this new `Outputs` type. See updated PR please 🙏